src/Agent.Worker/ExecutionContext.cs

```diff
@@ -396,6 +400,10 @@ public void InitializeJob(Pipelines.AgentJobRequestMessage message, Cancellation
             // Repositories
             Repositories = message.Resources.Repositories;
 
+            // JobSettings
+            JobSettings = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            JobSettings[WellKnownJobSettings.HasMultipleCheckouts] = message.Steps?.Where(x => Pipelines.PipelineConstants.IsCheckoutTask(x)).Count() > 1 ? Boolean.TrueString : Boolean.FalseString;
```

Comment:
Does this get set correctly if Steps is null? Should never happen.

Reply:
The compile works, which suggests that it can evaluate the boolean conditional even when steps is null. I assumed that would evaluate to false which is what we would want, but I can double check.